src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -205,18 +192,23 @@ public WebApplication Build()
                 // Drop the reference to the existing collection and set the inner collection
                 // to the new one. This allows code that has references to the service collection to still function.
                 _services.InnerCollection = services;
+
+                // Make builder.Configuration match the final configuration. To do that, we add the additional
+                // providers in the inner _hostBuilders's Configuration to the ConfigurationManager.
+                foreach (var provider in ((IConfigurationRoot)context.Configuration).Providers)
```

Comment:
I love this! There are 1 edge case we should handle, clearing providers. If the web application factory clears providers then clear providers on the configuration manager

The other is too edgy: somebody replaces `context.Configuration` with another implementation that doesn't implement `IConfigurationRoot`.